android/src/main/java/com/thunderhead/one_sdk_flutter/OnePlugin.java

```diff
@@ -29,132 +40,271 @@
-    final OneRequest sendInteractionRequest = new OneRequest.Builder()
-            .interactionPath(new OneInteractionPath(URI.create(interactionPath)))
-            .properties(properties)
-            .build();
-
-    final OneCall sendInteractionCall = One.sendInteraction(sendInteractionRequest);
-    sendInteractionCall.enqueue(new OneCallback() {
-      @Override
-      public void onSuccess(final OneResponse response) {
-        One.processResponse(response);
-        result.success(response.getTid());
-      }
-
-      @Override
-      public void onError(OneSDKError error) {
-        result.error(Integer.toString(error.getSystemCode()), error.getLocalizedMessage(), error);
-        Log.e(LOG_TAG, "SDK Error: " +  error.getErrorMessage());
-      }
-
-      @Override
-      public void onFailure(OneAPIError error) {
-        result.error(Integer.toString(error.getHttpStatusCode()), error.getLocalizedMessage(), error);
-        Log.e(LOG_TAG, "Api Error: " +  error.getErrorMessage());
-      }
-    });
-  }
+    private void optOut(MethodCall call, final Result result) {
+        OneOptOutConfiguration optOutConfiguration;
+
+        final HashMap<String, Object> arguments = call.arguments();
+        Boolean optOut = (Boolean) arguments.get("optOut");
+        List<String> options = (List<String>) arguments.get("options");
+
+        if (optOut == null) {
+            optOut = false;
+        }
+
+        if (options != null && !options.isEmpty()) {
+            for (String optOutOption : options) {
+                if (optOutOption.equals("cityCountryDetection")) {
+                    Set<OneOptInOptions> optInOptions = EnumSet.noneOf(OneOptInOptions.class);
+                    optInOptions.add(OneOptInOptions.CITY_COUNTRY_DETECTION);
+                    optOutConfiguration = new OneOptOutConfiguration.Builder()
+                            .optOut(optOut)
+                            .optInOptions(optInOptions)
+                            .build();
+                    One.setOptOutConfiguration(optOutConfiguration);
+                }
+                if (optOutOption.equals("allTracking")) {
+                    // Configure all opt-out settings.
+
+                    // Configures opt-out settings for tracking.
+                    optOutConfiguration = new OneOptOutConfiguration.Builder()
+                            .optOut(optOut)
+                            .build();
+                    One.setOptOutConfiguration(optOutConfiguration);
+
+                    // Configures opt-out settings for other options.
+                    Set<OneOptInOptions> allOptInOptions = EnumSet.allOf(OneOptInOptions.class);
+                    optOutConfiguration = new OneOptOutConfiguration.Builder()
+                            .optOut(optOut)
+                            .optInOptions(allOptInOptions)
+                            .build();
+                    One.setOptOutConfiguration(optOutConfiguration);
```

Comment:
This will override the previous call. This whole section is dubious. please refactor.

Reply:
You only need one opt out configuration and one call to `setOptOutconfiguration`